stistools/splice.py

```diff
@@ -0,0 +1,888 @@
+    overlap_pair_sections = []
+    overlap_trio_sections = []
+    n_pairs = len(first_pair)
+    n_trios = len(first_trio)
+    for i in range(n_pairs):
+        overlap_pair_sections.append([first_pair[i], second_pair[i]])
+    if n_trios > 0:
+        for i in range(n_trios):
+            overlap_trio_sections.append([first_trio[i], second_trio[i],
+                                          third_trio[i]])
+
+    return unique_sections, overlap_pair_sections, overlap_trio_sections
+
+
+# Merge overlapping sections
+def merge_overlap(overlap_sections,
+                  acceptable_dq_flags=(0, 64, 128, 1024, 2048),
+                  weight='sensitivity'):
+    """
+    Merges overlapping spectral regions. The basic workflow of this function
+    is to interpolate the sections into a common wavelength table and calculate
+    the weighted mean flux for each wavelength bin. If the fluxes are
+    inconsistent between each other, the code can use the flux with higher SNR
+    instead of the mean. If there are still outlier fluxes (compared to
+    neighboring pixels), the code uses the flux from the lower SNR section
+    instead. Co-added (merged) pixels will have their DQ flag set to `32768` if
+    they are the result of combining good pixels (according to the list of
+    acceptable flags). Their DQ flag will be set to `65536` if the combined
+    pixels do not have an acceptable DQ flag.
+
+    Parameters
+    ----------
+    overlap_sections : ``list``
+        List of dictionaries containing the overlapping spectra of neighboring
+        orders.
+
+    acceptable_dq_flags : array-like, optional
+        Data-quality flags that are acceptable when co-adding overlapping
+        spectra. The default values are (0, 64, 128, 1024, 2048), which
+        correspond to: 0 = regular pixel, 64 = vignetted pixel, 128 = pixel in
+        overscan region, 1024 = small blemish, 2048 = more than 30% of
+        background pixels rejected by sigma-clipping in the data reduction.
+
+    weight : ``str``, optional
+        Defines how to merge the overlapping sections. The options currently
+        implemented are ``'sensitivity'`` and ``'snr'`` (inverse square of the
+        uncertainties). Default is ``'sensitivity'``.
+
+    Returns
+    -------
+    overlap_merged : ``dict``
+        Dictionary containing the merged overlapping spectrum.
+    """
+    n_overlaps = len(overlap_sections)
+    bitwise_or_acceptable_dq_flags = reduce(np.bitwise_or, acceptable_dq_flags)
+
+    # First we need to determine which spectrum has a higher sensitivity
+    avg_sensitivity = np.array([np.nanmean(ok['net'] / ok['flux'])
+                                for ok in overlap_sections])
+
+    # We interpolate the lower-SNR spectra to the wavelength bins of the higher
+    # SNR spectrum.
+    max_sens_idx = np.where(avg_sensitivity == np.nanmax(avg_sensitivity))[0][0]
+    overlap_ref = overlap_sections.pop(max_sens_idx)
```

Comment:
This has the side effect of modifying the input variable.  You should probably make a copy inside the function instead.